components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -644,13 +656,17 @@ export function createWalletApi (
           baseQuery
         ) => {
           try {
-            const { jsonRpcService } = baseQuery(undefined).data
+            const { braveWalletService, jsonRpcService }
+              = baseQuery(undefined).data
+            const { originInfo } = await braveWalletService.getActiveOrigin()
 
             await dispatch(
               walletApi.endpoints.setSelectedCoin.initiate(coin)
             ).unwrap()
 
-            const { success } = await jsonRpcService.setNetwork(chainId, coin)
+            const { success }
+              = await jsonRpcService.setNetwork(chainId, coin,
+                                                originInfo.origin)
```

Comment:
Same here. This could be jsonRpcService.setNetworkForActiveOrigin(...)

Reply:
fixed in 210bbec4829bea9ca6ae39e73d2bbd6bb233ad9e and 4ebbe1249cb16e6694ddefe20b5c7505a265bcc3